### PR TITLE
[Do Not Merge] CORTX-29578: Design for new VKVV leaf node

### PR DESCRIPTION
**Pull request by ydb242**
Signed-off-by: Yash Bhamare <yash.bhamare@seagate.com>

# Problem Statement
- CORTX-29578: Design for new VKVV leaf node

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: btree/btree.c [hunk 1]
```diff
@@ -4324,6 +4324,125 @@ static void fkvv_rec_del_credit(const struct nd *node, m0_bcount_t ksize,
  * change for most of the records.
  *
  */
+
+/**
+ * @brief Proposed design to use benefits of indirect addressing and embedding
+ *        records in nodes of VKVV type.
+ *
+ * We will leverage the benefits of current design such as floating directory,
+ * seperation of keys and values, and add indirect addressing the current
```
Comment (shashank-parulekar) [inline btree/btree.c:4333]: s/addressing the/addressing in the/

File: btree/btree.c [hunk 2]
```diff
@@ -4324,6 +4324,125 @@ static void fkvv_rec_del_credit(const struct nd *node, m0_bcount_t ksize,
...
+ * The view of the node will be:
+ *
+ *                +--------------------+
+ *                |                    |
+ *           +----+----------------+   |
+ *           |    |                |   |
+ *        +--+----+------------+   |   |
+ *        |  |    |            |   |   |
+ * +------v--v----v---+------+-+-+-+-+-+-+---+-------+---+----+--+
+ * |      |  |    |   |      |   |   |   |   |       |   |    |  |
```
Comment (shashank-parulekar) [inline btree/btree.c:4365]: Add some variable denoting the key offset and value offset in the directory.

**Review by upendrapatwardhan — APPROVED** [review]

**Review by shashank-parulekar — CHANGES_REQUESTED** [review]

File: btree/btree.c [hunk 3]
```diff
@@ -4324,6 +4324,128 @@ static void fkvv_rec_del_credit(const struct nd *node, m0_bcount_t ksize,
...
+ *                |                    |
+ *           +----+----------------+   |
+ *           |    |                |   |
+ *        +--+----+------------+   |   |
+ *        |  |    |            |   |   |
+ * +------v--v----v---+------+-+-+-+-+-+-+---+-------+---+----+--+
+ * |      |  |    |   |      |ok0|ok1|ok2|   |       |   |    |  |
+ * |      |  |    |   |      +---+---+---+---+       |   |    |  |
+ * |      |  |    |   |      |sk0|sk1|sk2|   |       |   |    |  |
+ * | Node |K0| K1 | K2|      +---+---+---+---+       |V0 | V1 |V0|
```
Comment (shashank-parulekar) [inline btree/btree.c:4368]: s/V0/V2/1

**Review by shashank-parulekar — APPROVED** [review]

**Comment by ydb242** [comment]
The design will be kept in local branch till further use.

**Pull request closed without merging**